src/fs.rs

```diff
@@ -695,12 +716,12 @@ pub(crate) struct FsIoAdapter<'a, IO: ReadWriteSeek, TP, OCC> {
 }
 
 impl<IO: ReadWriteSeek, TP, OCC> IoBase for FsIoAdapter<'_, IO, TP, OCC> {
-    type Error = IO::Error;
+    type Error = Error<IO::Error>;
```

Comment:
why this change? I see some more changes to error handling but I don't understand how it is related to clearing dirty flags

Reply:
I don't quite remember exactly. Somewhere the errors types were not compatible during propagation.